api/src/main/java/me/lokka30/treasury/api/economy/account/BankAccount.java

```diff
@@ -24,89 +24,137 @@
 
     /**
      * Make a user a member of the bank.
-     *
-     * @author lokka30
      * @param memberId the {@link UUID} of the new member
      * @param subscription the {@link EconomySubscriber} accepting whether the member was added
+     * @author lokka30
      * @since v1.0.0
      */
     void addBankMember(@NotNull UUID memberId, @NotNull EconomySubscriber<Boolean> subscription);
 
     /**
      * Make a user an owner of the bank.
-     *
-     * @author lokka30
      * @param ownerId the {@link UUID} of the potential new owner
      * @param subscription the {@link EconomySubscriber} accepting whether the owner was added
+     * @author lokka30
      * @since v1.0.0
      */
     void addBankOwner(@NotNull UUID ownerId, @NotNull EconomySubscriber<Boolean> subscription);
 
     /**
      * Remove a member of the bank.
-     *
-     * @author lokka30
      * @param memberId the {@link UUID} of the member removed
      * @param subscription the {@link EconomySubscriber} accepting whether the member was removed
+     * @author lokka30
      * @since v1.0.0
      */
     void removeBankMember(@NotNull UUID memberId, @NotNull EconomySubscriber<Boolean> subscription);
 
     /**
      * Remove an owner of the bank.
-     *
-     * @author lokka30
      * @param ownerId the {@link UUID} of the owner removed
      * @param subscription the {@link EconomySubscriber} accepting whether the owner was removed
+     * @author lokka30
      * @since v1.0.0
      */
     void removeBankOwner(@NotNull UUID ownerId, @NotNull EconomySubscriber<Boolean> subscription);
+
+    /**
+     * Checks whether given player has the given permission on this bank account.
+     * @param player the {@link UUID} of the player to check if they have the permission
+     * @param permission the permission to check
+     * @param subscription the {@link EconomySubscriber} accepting whether the player has the permission
+     * @author MrNemo64
+     * @see #canConsultBalance(UUID, EconomySubscriber) 
+     * @see #canWithdraw(UUID, EconomySubscriber) 
+     * @see #canDeposit(UUID, EconomySubscriber)
+     * @see BankAccountPermission
+     * @since v1.0.0
+     */
+    void hasPermission(@NotNull UUID player, @NotNull BankAccountPermission permission, @NotNull EconomySubscriber<Boolean> subscription);
+
+    /**
+     * Checks whether the given player can consult the balance of this bank account,
+     * in other words, if they have the {@link BankAccountPermission#CONSULT} permission.
+     * @param player the {@link UUID} of the player to check if they can consult the balance
+     * @param subscription the {@link EconomySubscriber} accepting whether the player can consult the balance
+     * @author MrNemo64
+     * @see #hasPermission(UUID, BankAccountPermission, EconomySubscriber)
+     * @see BankAccountPermission#CONSULT
+     * @since v1.0.0
+     */
+    default void canConsultBalance(@NotNull UUID player, @NotNull EconomySubscriber<Boolean> subscription) {
```

Comment:
I don't believe we should have the methods for individual permissions - rather, just the one that asks 'has permission' with specified enum constant.

Reply:
@MrNemo64 you marked this as resolved but didn't leave a comment - what do you think about this?